sdcm/utils/file.py

```diff
@@ -0,0 +1,109 @@
+from typing import Optional, TextIO, List, Tuple, Union, AnyStr
+from re import Pattern, compile, IGNORECASE
+from sdcm.sct_events import DatabaseLogEvent
+
+
+class File:
+    """
+    File object that support chaining and context managing
+
+    Example:
+        >>> assert File('/tmp/123', 'r+').writelines(['someline\\n', 'someline #1\\n', 'someline #2\\n']
+        >>> ).move_to_begging().readlines() == ['someline\\n', 'someline #1\\n', 'someline #2\\n']
+    """
+    def __init__(self, path: str, mode: str = 'r', buffering: Optional[int] = None,
+                 encoding: Optional[str] = None, errors: Optional[str] = None, newline: Optional[str] = None,
+                 closefd: bool = True):
+        self.path = path
+        self.mode = mode
+        self.buffering = buffering
+        self.encoding = encoding
+        self.errors = errors
+        self.newline = newline
+        self.closefd = closefd
+        self._io = self._open()
+
+    def get_file_length(self) -> int:
+        return self._open().seek(0, 2)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._io.close()
+
+    def _open(self) -> TextIO:
+        kwargs = {attr_name: getattr(self, attr_name) for attr_name in
+                  ['mode', 'buffering', 'encoding', 'errors', 'closefd'] if getattr(self, attr_name, None) is not None}
+        return open(self.path, **kwargs)  # deepcode ignore missing~close~open: <comment the reason here>
+
+    def move_to(self, pos) -> 'File':
+        self._io.seek(pos)
+        return self
+
+    def move_to_end(self) -> 'File':
+        self._io.seek(0, 2)
+        return self
+
+    def move_to_begging(self) -> 'File':
+        self._io.seek(0)
+        return self
+
+    def move_to_relative(self, pos: int) -> 'File':
+        self._io.seek(pos, 1)
+        return self
+
+    def move_to_relative_from_end(self, pos: int = 0) -> 'File':
+        self._io.seek(pos, 2)
+        return self
+
+    def write(self, s: AnyStr) -> 'File':
+        self._io.write(s)
+        return self
+
+    def writelines(self, s: List[AnyStr]) -> 'File':
+        self._io.writelines(s)
+        return self
+
+    def seek(self, offset: int, whence: int = 0) -> 'File':
+        self._io.seek(offset, whence)
+        return self
+
+    def read(self, n: int = -1) -> AnyStr:
+        return self._io.read(n)
+
+    def readline(self, limit: int = -1) -> AnyStr:
+        return self._io.readline(limit)
+
+    def readlines(self, hint: int = -1) -> List[AnyStr]:
+        return self._io.readlines(hint)
+
+    def read_lines_filtered(self, *patterns: Union[str, Pattern, DatabaseLogEvent], ignore_case=True) \
```

Comment:
I think it would benefit from having markers, like in dtest.

Since in most cases we don't want to read the whole log, just from a certain position

Reply:
I thought about it, but since you can address certain position by simple number via seek, i decided to settle on move_to* routines.

Reply:
But you are not using them in the PR

Reply:
I'am doing it here:
        if not start_from_beginning:
            stream.move_to_end()